climakitae/threshold_tools.py

```diff
@@ -0,0 +1,613 @@
+                    return_event = 1.0 - (1.0 / return_period)
+                    return_value = fitted_distr.ppf(return_event)
+                    return_value = round(return_value, 5)
+                except ValueError:
+                    return_value = np.nan
+
+            if distr == "gumbel":
+                try:
+                    lmoments = ldistr.gum.lmom_fit(ams)
+                    fitted_distr = stats.gumbel_r(**lmoments)
+                    return_event = 1.0 - (1.0 / return_period)
+                    return_value = fitted_distr.ppf(return_event)
+                    return_value = round(return_value, 5)
+                except ValueError:
+                    return_value = np.nan
+
+            if distr == "weibull":
+                try:
+                    lmoments = ldistr.wei.lmom_fit(ams)
+                    fitted_distr = stats.weibull_min(**lmoments)
+                    return_event = 1.0 - (1.0 / return_period)
+                    return_value = fitted_distr.ppf(return_event)
+                    return_value = round(return_value, 5)
+                except ValueError:
+                    return_value = np.nan
+
+            if distr == "pearson3":
+                try:
+                    lmoments = ldistr.pe3.lmom_fit(ams)
+                    fitted_distr = stats.pearson3(**lmoments)
+                    return_event = 1.0 - (1.0 / return_period)
+                    return_value = fitted_distr.ppf(return_event)
+                    return_value = round(return_value, 5)
+                except ValueError:
+                    return_value = np.nan
+
+            if distr == "genpareto":
+                try:
+                    lmoments = ldistr.gpa.lmom_fit(ams)
+                    fitted_distr = stats.genpareto(**lmoments)
+                    return_event = 1.0 - (1.0 / return_period)
+                    return_value = fitted_distr.ppf(return_event)
+                    return_value = round(return_value, 5)
+                except ValueError:
+                    return_value = np.nan
+
+            return return_value
+
+        return xr.apply_ufunc(
+            function,
+            ams,
+            input_core_dims=[["time"]],
+            exclude_dims=set(("time",)),
+            output_core_dims=[[]],
+        )
+
+    return_value = apply_function(ams)
+
+    return_value = return_value.rename("return_value")
+    new_ds = return_value.to_dataset()
+
+    if multiple_points == True:
+        new_ds = new_ds.unstack("allpoints")
+
+    new_ds["return_value"].attrs["return period"] = "1 in {} year event".format(
+        str(return_period)
+    )
+    new_ds.attrs = ams_attributes
+    new_ds.attrs["distribution"] = "{}".format(str(distr))
+
+    return new_ds
+
+
+#####################################################################
+
+
+def get_return_prob(ams, threshold, distr="gev", multiple_points=True):
+
+    distrs = ["gev", "gumbel", "weibull", "pearson3", "genpareto"]
```

Comment:
I'm wondering if we should add a note in the notebook describing when each of these distributions is most commonly used?  Like a standard best practice.

Reply:
yeah! i think this might be helpful to include in the documentation, where maybe we can add more details about when certain distributions are used.